toolkit/metric/api.go

```diff
@@ -18,16 +18,21 @@
 package metric
 
 // NewCounter creates a new counter metrics.
-func NewCounter(name string, opt ...MeterOpt) *CounterRef {
+func NewCounter(name string, opt ...interface{}) *CounterRef {
```

Comment:
Why we need change the opt as interface? If it necessary, please add comment in the method to describe what kind of data we need.

Reply:
Using the `interface{}` type will allow us to handle it better in the interceptor, at least we don't need to do heavy type conversion tasks, which will bring a lot of unnecessary work.

I will strive to write detailed comments to explain to users why and how we should use it.

Reply:
`interface{}` seems a very general type. For a counter, the parameters should be very certain. And building a counter is not a high-frequency operation, I can't see why change this.

Reply:
But it's not easy understanding in the user, right? 
Can we add an opt structure, and using the enhance instance as the option configuration. Then, user can know which kind of opts they can fill the arguments.

Reply:
> But it's not easy understanding in the user, right?

Not only this, more about. How to prevent the wrong implementation in the method? A certain type is a good thing, we should not expose an `interface{}` but no other condition, unless this is the use case.

Reply:
yeah, I used `MeterOpt` as a type constraint to tell users how to use it.
@wu-sheng @mrproliu